src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+/*
+ * Trident - A Multithreaded Server Alternative
+ * Copyright 2016 The TridentSDK Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.tridentsdk.chat;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import lombok.*;
+import lombok.experimental.Accessors;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Represents a component in a Minecraft chat format.
+ */
+@Accessors(chain = true)
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class ChatComponent {
+
+    @Getter @Setter
+    private String text, translate, selector, insertion, scoreUsername, scoreObjective;
+
+    @Getter @Setter
+    private ChatColor color;
+
+    @Getter @Setter
+    private ClickEvent clickEvent;
+
+    @Getter @Setter
+    private HoverEvent hoverEvent;
+
+    private List<JsonElement> with = Lists.newCopyOnWriteArrayList();
+    private List<ChatComponent> extra = Lists.newCopyOnWriteArrayList();
+    private AtomicBoolean bold, italic, underlined, strikethrough, obfuscated;
```

Comment:
THREAD-SAFETY
This is a poor construct to use for a thread-safe class. It doesn't save (that much) memory, and can be solved by using a volatile boolean set to false as a default value instead.